lib/scheduler.js

```diff
@@ -159,7 +159,7 @@ class Scheduler {
                     return;
                 }
                 if (schedule.period.years > 1) {
-                    const diff = Math.floor(this.monthDiff(this.period.fromDate, new Date(context.now)) / 12);
+                    const diff = Math.floor(this.monthDiff(schedule.period.fromDate, new Date(context.now)) / 12);
```

Comment:
All the other branches use `schedule.period.fromDate`, so this is most likely a typo